lib/fileio/jobs/readJob.cpp

```diff
@@ -103,7 +102,22 @@ UsdMaya_ReadJob::Read(std::vector<MDagPath>* addedDagPaths)
 
     // Layer and Stage used to Read in the USD file
     UsdStageCacheContext stageCacheContext(UsdMayaStageCache::Get());
-    UsdStageRefPtr stage = UsdStage::Open(rootLayer, sessionLayer);
+    UsdStageRefPtr stage;
+    if (mImportData.empty())
```

Comment:
Change to how we open the stage depending on what the import data contains.

Reply:
Yes correct, now that I've moved the filename into the import data it won't be empty. I've changed the stage open as you've suggested.